### ISSUE TITLE
Abnormal recovery problem when GPU node fails

### ISSUE THREAD

**Issue by fyf2016** [NVIDIA/nccl#989]
@sjeaugey ，Almighty sjeaugey, I have a question related to NCCL failure recovery: When I use double binary tree as the communication algorithm, if a GPU node in the cluster fails, will the entire NCCL communication process be interrupted? If interrupted, does NCCL have a corresponding recovery method?

**Comment by sjeaugey** [comment]
Yes, a GPU failures will cause the NCCL communicator to be broken. You will need to call `ncclCommAbort` on all GPUs which are still in good condition, then recreate a new communicator with the remaining GPUs (or even new ones).

Note this has nothing to do with the Tree algorithm; it is the general way fault handling is working in NCCL.

**Comment by fyf2016** [comment]
> Yes, a GPU failures will cause the NCCL communicator to be broken. You will need to call `ncclCommAbort` on all GPUs which are still in good condition, then recreate a new communicator with the remaining GPUs (or even new ones).
> 
> Note this has nothing to do with the Tree algorithm; it is the general way fault handling is working in NCCL.

@sjeaugey ，Another question is, after I call ncclCommAbort, does the process of using the remaining nodes (or even new ones) to create a new communicator require the programmer to create it manually, or will NCCL automatically detect the topology to complete the creation?

**Comment by sjeaugey** [comment]
Unfortunately this is not something NCCL as a mere communication library can do.

The application needs to have a process management system which will detect the faults, make sure all ranks share the same awareness of faults, and keep tabs on which ranks are available to continue (or even work spawn new ones). This is a whole field in itself; reliable fault detection and management is a complex subject which has has been studied for decades in the HPC world.

NCCL is just providing the part communication libraries need to provide in the global picture: report communication faults, then allow for tear down and the re-creation of a new group.

**Comment by stephenmsachs** [comment]
### Issue Cleanup: Helping Us Focus on Current Challenges

  We're [reviewing](https://github.com/NVIDIA/nccl/discussions/1761) older issues to ensure we prioritize the most relevant and active ones. Since this issue hasn't seen updates in over 6 months, we'll be closing it for now.

  ,*This change helps us focus our efforts on addressing any current issues our users are facing.* If this issue still affects you, please don't hesitate to reopen it with a quick update (e.g., "Still relevant on [version=X]").
  Thanks for your understanding and for contributing to NCCL.
